### ISSUE TITLE
couldn't install latest version using composer

### ISSUE THREAD

**Issue by ooples** [cloudflare/Cloudflare-Magento#48]
I tried following the readme instructions for installing by using composer and it failed. Here is what I did

![composer_failed](https://cloud.githubusercontent.com/assets/3075685/24611373/e1191c98-184f-11e7-8b83-90d12a6fd4aa.png)

**Comment by jwineman** [comment]
Did you the plugin installed previously and you were trying to upgrade it?

**Comment by ooples** [comment]
Yes I was trying to upgrade it but it won't upgrade

**Comment by jwineman** [comment]
If you're upgrading instead of installing I think you want to run:
```
composer update cloudflare/cloudflare-magento
```

Does that work?

**Comment by ooples** [comment]
When I do that command it says nothing to install or update even though I haven't been able to update yet

**Comment by jwineman** [comment]
What is the version of `cloudflare/cloudflare-magento` in your `composer.json`?

**Comment by ooples** [comment]
It says require "cloudflare/cloudflare-magento": "^1.0"

**Comment by jwineman** [comment]
Try this:

1. Delete that line in your `composer.json`
2. Run `composer update`
3. Run `composer require cloudflare/cloudflare-magento`

**Comment by ooples** [comment]
I ran those commands and I didn't get any errors but I also don't see cloudflare on my admin section anywhere. Where am I supposed to be looking for the cloudflare section?

**Comment by jwineman** [comment]
Did you run `bin/magento setup:upgrade` after reinstalling with composer?

**Comment by ooples** [comment]
Yes and everything looks like it worked but I don't see cloudflare settings anywhere. Where should I be looking?

**Comment by jwineman** [comment]
Do you see it under "System" > "Integrations"?  If not what if you run:
```
bin\magento module:enable CloudFlare_Plugin
```

**Comment by ooples** [comment]
This is what happened after I enabled the plugin. Another error :(

[2017-04-03 23:22:58] main.CRITICAL: Exception: Warning: json_decode() expects parameter 1 to be string, object given in /var/www/vhosts/ooples.com/httpdocs/vendor/cloudflare/cloudflare-plugin-backend/src/Integration/DefaultConfig.php on line 14 in /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/App/ErrorHandler.php:61
Stack trace:
#0 [internal function]: Magento\Framework\App\ErrorHandler->handler(2, 'json_decode() e...', '/var/www/vhosts...', 14, Array)
#1 /var/www/vhosts/ooples.com/httpdocs/vendor/cloudflare/cloudflare-plugin-backend/src/Integration/DefaultConfig.php(14): json_decode(Object(Magento\Framework\ObjectManager\ObjectManager), true)
#2 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php(93): CF\Integration\DefaultConfig->__construct(Object(Magento\Framework\ObjectManager\ObjectManager))
#3 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(88): Magento\Framework\ObjectManager\Factory\AbstractFactory->createObject('CF\\Integration\\...', Array)
#4 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(130): Magento\Framework\ObjectManager\Factory\Compiled->create('CF\\Integration\\...')
#5 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(67): Magento\Framework\ObjectManager\Factory\Compiled->get('CF\\Integration\\...')
#6 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(130): Magento\Framework\ObjectManager\Factory\Compiled->create('CloudFlare\\Plug...')
#7 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(67): Magento\Framework\ObjectManager\Factory\Compiled->get('CloudFlare\\Plug...')
#8 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(130): Magento\Framework\ObjectManager\Factory\Compiled->create('CloudFlare\\Plug...')
#9 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(67): Magento\Framework\ObjectManager\Factory\Compiled->get('CloudFlare\\Plug...')
#10 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(130): Magento\Framework\ObjectManager\Factory\Compiled->create('CloudFlare\\Plug...')
#11 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(67): Magento\Framework\ObjectManager\Factory\Compiled->get('CloudFlare\\Plug...')
#12 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/ObjectManager.php(71): Magento\Framework\ObjectManager\Factory\Compiled->create('CloudFlare\\Plug...')
#13 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Event/ObserverFactory.php(33): Magento\Framework\ObjectManager\ObjectManager->get('CloudFlare\\Plug...')
#14 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Event/Invoker/InvokerDefault.php(59): Magento\Framework\Event\ObserverFactory->get('CloudFlare\\Plug...')
#15 /var/www/vhosts/ooples.com/httpdocs/var/generation/Magento/Framework/Event/Invoker/InvokerDefault/Interceptor.php(24): Magento\Framework\Event\Invoker\InvokerDefault->dispatch(Array, Object(Magento\Framework\Event\Observer))
#16 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Event/Manager.php(66): Magento\Framework\Event\Invoker\InvokerDefault\Interceptor->dispatch(Array, Object(Magento\Framework\Event\Observer))
#17 /var/www/vhosts/ooples.com/httpdocs/var/generation/Magento/Framework/Event/Manager/Proxy.php(95): Magento\Framework\Event\Manager->dispatch('clean_cache_by_...', Array)
#18 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Model/AbstractModel.php(801): Magento\Framework\Event\Manager\Proxy->dispatch('clean_cache_by_...', Array)
#19 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Model/ResourceModel/Db/AbstractDb.php(818): Magento\Framework\Model\AbstractModel->afterSave()
#20 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Model/ResourceModel/Db/AbstractDb.php(408): Magento\Framework\Model\ResourceModel\Db\AbstractDb->processAfterSaves(Object(Magestore\Webpos\Model\Staff\WebPosSession))
#21 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Model/AbstractModel.php(631): Magento\Framework\Model\ResourceModel\Db\AbstractDb->save(Object(Magestore\Webpos\Model\Staff\WebPosSession))
#22 /var/www/vhosts/ooples.com/httpdocs/app/code/Magestore/Webpos/Helper/Permission.php(155): Magento\Framework\Model\AbstractModel->save()
#23 /var/www/vhosts/ooples.com/httpdocs/app/code/Magestore/Webpos/Controller/Rest.php(42): Magestore\Webpos\Helper\Permission->authorizeSession('5garb55fo9ntht9...')
#24 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/module-webapi/Controller/Rest.php(334): Magestore\Webpos\Controller\Rest->checkPermissions()
#25 /var/www/vhosts/ooples.com/httpdocs/app/code/Magestore/Webpos/Controller/Rest.php(94): Magento\Webapi\Controller\Rest->validateRequest()
#26 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/module-webapi/Controller/Rest.php(216): Magestore\Webpos\Controller\Rest->processApiRequest()
#27 /var/www/vhosts/ooples.com/httpdocs/var/generation/Magestore/Webpos/Controller/Rest/Interceptor.php(24): Magento\Webapi\Controller\Rest->dispatch(Object(Magento\Framework\App\Request\Http))
#28 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/App/Http.php(135): Magestore\Webpos\Controller\Rest\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#29 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#30 /var/www/vhosts/ooples.com/httpdocs/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#31 {main}

Next Exception: Report ID: webapi-58e2d9529e110; Message: Warning: json_decode() expects parameter 1 to be string, object given in /var/www/vhosts/ooples.com/httpdocs/vendor/cloudflare/cloudflare-plugin-backend/src/Integration/DefaultConfig.php on line 14 in /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Webapi/ErrorProcessor.php:195
Stack trace:
#0 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Webapi/ErrorProcessor.php(139): Magento\Framework\Webapi\ErrorProcessor->_critical(Object(Exception))
#1 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/module-webapi/Controller/Rest.php(219): Magento\Framework\Webapi\ErrorProcessor->maskException(Object(Exception))
#2 /var/www/vhosts/ooples.com/httpdocs/var/generation/Magestore/Webpos/Controller/Rest/Interceptor.php(24): Magento\Webapi\Controller\Rest->dispatch(Object(Magento\Framework\App\Request\Http))
#3 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/App/Http.php(135): Magestore\Webpos\Controller\Rest\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#4 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#5 /var/www/vhosts/ooples.com/httpdocs/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#6 {main} [] []
[2017-04-03 23:22:58] main.CRITICAL: Exception: Warning: json_decode() expects parameter 1 to be string, object given in /var/www/vhosts/ooples.com/httpdocs/vendor/cloudflare/cloudflare-plugin-backend/src/Integration/DefaultConfig.php on line 14 in /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/App/ErrorHandler.php:61
Stack trace:
#0 [internal function]: Magento\Framework\App\ErrorHandler->handler(2, 'json_decode() e...', '/var/www/vhosts...', 14, Array)
#1 /var/www/vhosts/ooples.com/httpdocs/vendor/cloudflare/cloudflare-plugin-backend/src/Integration/DefaultConfig.php(14): json_decode(Object(Magento\Framework\ObjectManager\ObjectManager), true)
#2 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php(93): CF\Integration\DefaultConfig->__construct(Object(Magento\Framework\ObjectManager\ObjectManager))
#3 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(88): Magento\Framework\ObjectManager\Factory\AbstractFactory->createObject('CF\\Integration\\...', Array)
#4 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(130): Magento\Framework\ObjectManager\Factory\Compiled->create('CF\\Integration\\...')
#5 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(67): Magento\Framework\ObjectManager\Factory\Compiled->get('CF\\Integration\\...')
#6 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(130): Magento\Framework\ObjectManager\Factory\Compiled->create('CloudFlare\\Plug...')
#7 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(67): Magento\Framework\ObjectManager\Factory\Compiled->get('CloudFlare\\Plug...')
#8 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(130): Magento\Framework\ObjectManager\Factory\Compiled->create('CloudFlare\\Plug...')
#9 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(67): Magento\Framework\ObjectManager\Factory\Compiled->get('CloudFlare\\Plug...')
#10 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(130): Magento\Framework\ObjectManager\Factory\Compiled->create('CloudFlare\\Plug...')
#11 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(67): Magento\Framework\ObjectManager\Factory\Compiled->get('CloudFlare\\Plug...')
#12 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/ObjectManager/ObjectManager.php(71): Magento\Framework\ObjectManager\Factory\Compiled->create('CloudFlare\\Plug...')
#13 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Event/ObserverFactory.php(33): Magento\Framework\ObjectManager\ObjectManager->get('CloudFlare\\Plug...')
#14 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Event/Invoker/InvokerDefault.php(59): Magento\Framework\Event\ObserverFactory->get('CloudFlare\\Plug...')
#15 /var/www/vhosts/ooples.com/httpdocs/var/generation/Magento/Framework/Event/Invoker/InvokerDefault/Interceptor.php(24): Magento\Framework\Event\Invoker\InvokerDefault->dispatch(Array, Object(Magento\Framework\Event\Observer))
#16 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Event/Manager.php(66): Magento\Framework\Event\Invoker\InvokerDefault\Interceptor->dispatch(Array, Object(Magento\Framework\Event\Observer))
#17 /var/www/vhosts/ooples.com/httpdocs/var/generation/Magento/Framework/Event/Manager/Proxy.php(95): Magento\Framework\Event\Manager->dispatch('clean_cache_by_...', Array)
#18 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Model/AbstractModel.php(801): Magento\Framework\Event\Manager\Proxy->dispatch('clean_cache_by_...', Array)
#19 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Model/ResourceModel/Db/AbstractDb.php(818): Magento\Framework\Model\AbstractModel->afterSave()
#20 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Model/ResourceModel/Db/AbstractDb.php(408): Magento\Framework\Model\ResourceModel\Db\AbstractDb->processAfterSaves(Object(Magestore\Webpos\Model\Staff\WebPosSession))
#21 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Model/AbstractModel.php(631): Magento\Framework\Model\ResourceModel\Db\AbstractDb->save(Object(Magestore\Webpos\Model\Staff\WebPosSession))
#22 /var/www/vhosts/ooples.com/httpdocs/app/code/Magestore/Webpos/Helper/Permission.php(155): Magento\Framework\Model\AbstractModel->save()
#23 /var/www/vhosts/ooples.com/httpdocs/app/code/Magestore/Webpos/Controller/Rest.php(42): Magestore\Webpos\Helper\Permission->authorizeSession('5garb55fo9ntht9...')
#24 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/module-webapi/Controller/Rest.php(334): Magestore\Webpos\Controller\Rest->checkPermissions()
#25 /var/www/vhosts/ooples.com/httpdocs/app/code/Magestore/Webpos/Controller/Rest.php(94): Magento\Webapi\Controller\Rest->validateRequest()
#26 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/module-webapi/Controller/Rest.php(216): Magestore\Webpos\Controller\Rest->processApiRequest()
#27 /var/www/vhosts/ooples.com/httpdocs/var/generation/Magestore/Webpos/Controller/Rest/Interceptor.php(24): Magento\Webapi\Controller\Rest->dispatch(Object(Magento\Framework\App\Request\Http))
#28 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/App/Http.php(135): Magestore\Webpos\Controller\Rest\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#29 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#30 /var/www/vhosts/ooples.com/httpdocs/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#31 {main}

Next Exception: Report ID: webapi-58e2d952f296d; Message: Warning: json_decode() expects parameter 1 to be string, object given in /var/www/vhosts/ooples.com/httpdocs/vendor/cloudflare/cloudflare-plugin-backend/src/Integration/DefaultConfig.php on line 14 in /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Webapi/ErrorProcessor.php:195
Stack trace:
#0 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/Webapi/ErrorProcessor.php(139): Magento\Framework\Webapi\ErrorProcessor->_critical(Object(Exception))
#1 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/module-webapi/Controller/Rest.php(219): Magento\Framework\Webapi\ErrorProcessor->maskException(Object(Exception))
#2 /var/www/vhosts/ooples.com/httpdocs/var/generation/Magestore/Webpos/Controller/Rest/Interceptor.php(24): Magento\Webapi\Controller\Rest->dispatch(Object(Magento\Framework\App\Request\Http))
#3 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/App/Http.php(135): Magestore\Webpos\Controller\Rest\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#4 /var/www/vhosts/ooples.com/httpdocs/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#5 /var/www/vhosts/ooples.com/httpdocs/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#6 {main} [] []

**Comment by jwineman** [comment]
What if you run `bin/magento clean:cache`?

If that doesn't fix it what version of Magento are you running?

**Comment by ooples** [comment]
I tried that and it didn't fix it. I'm using magento 2.1.4

**Comment by jwineman** [comment]
Alright I'll try to reproduce and get back to you.

**Comment by jwineman** [comment]
Did you run `bin\magento setup:di:compile` after `setup:upgrade`?

**Comment by ooples** [comment]
Yes I did and then ran clean cache after that

**Comment by ooples** [comment]
Yes I did and I did all the steps over again just to be sure and I'm
getting the same exact error message

**Comment by jwineman** [comment]
Steps to reproduce:
```
bin/magento deploy:mode:set production
bin/magento setup:upgrade
```

**Comment by jwineman** [comment]
Hi @ooples,

This should be fixed in https://github.com/cloudflare/Cloudflare-Magento/releases/tag/v1.1.3.

Can you upgrade and let me know if the problem still persists?

**Comment by ooples** [comment]
So just to confirm, to upgrade I have to just run composer update again?

**Comment by ooples** [comment]
Yeah I upgraded and tried to turn on the extension and this is the error it
is giving me now

[2017-04-05 21:09:45] main.CRITICAL: Exception: Warning: json_decode()
expects parameter 1 to be string, object given in /var/www/vhosts/
ooples.com/httpdocs/vendor/cloudflare/cloudflare-plugin-backend/src/Integration/DefaultConfig.php
on line 14 in /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/App/ErrorHandler.php:61
Stack trace:
#0 [internal function]: Magento\Framework\App\ErrorHandler->handler(2,
'json_decode() e...', '/var/www/vhosts...', 14, Array)
#1 /var/www/vhosts/
ooples.com/httpdocs/vendor/cloudflare/cloudflare-plugin-backend/src/Integration/DefaultConfig.php(14):
json_decode(Object(Magento\Framework\ObjectManager\ObjectManager), true)
#2 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php(93):
CF\Integration\DefaultConfig->__construct(Object(Magento\Framework\ObjectManager\ObjectManager))
#3 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(88):
Magento\Framework\ObjectManager\Factory\AbstractFactory->createObject('CF\\Integration\\...',
Array)
#4 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(130):
Magento\Framework\ObjectManager\Factory\Compiled->create('CF\\Integration\\...')
#5 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(67):
Magento\Framework\ObjectManager\Factory\Compiled->get('CF\\Integration\\...')
#6 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(130):
Magento\Framework\ObjectManager\Factory\Compiled->create('CloudFlare\\Plug...')
#7 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(67):
Magento\Framework\ObjectManager\Factory\Compiled->get('CloudFlare\\Plug...')
#8 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(130):
Magento\Framework\ObjectManager\Factory\Compiled->create('CloudFlare\\Plug...')
#9 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(67):
Magento\Framework\ObjectManager\Factory\Compiled->get('CloudFlare\\Plug...')
#10 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(130):
Magento\Framework\ObjectManager\Factory\Compiled->create('CloudFlare\\Plug...')
#11 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/ObjectManager/Factory/Compiled.php(67):
Magento\Framework\ObjectManager\Factory\Compiled->get('CloudFlare\\Plug...')
#12 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/ObjectManager/ObjectManager.php(71):
Magento\Framework\ObjectManager\Factory\Compiled->create('CloudFlare\\Plug...')
#13 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/Event/ObserverFactory.php(33):
Magento\Framework\ObjectManager\ObjectManager->get('CloudFlare\\Plug...')
#14 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/Event/Invoker/InvokerDefault.php(59):
Magento\Framework\Event\ObserverFactory->get('CloudFlare\\Plug...')
#15 /var/www/vhosts/
ooples.com/httpdocs/var/generation/Magento/Framework/Event/Invoker/InvokerDefault/Interceptor.php(24):
Magento\Framework\Event\Invoker\InvokerDefault->dispatch(Array,
Object(Magento\Framework\Event\Observer))
#16 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/Event/Manager.php(66):
Magento\Framework\Event\Invoker\InvokerDefault\Interceptor->dispatch(Array,
Object(Magento\Framework\Event\Observer))
#17 /var/www/vhosts/
ooples.com/httpdocs/var/generation/Magento/Framework/Event/Manager/Proxy.php(95):
Magento\Framework\Event\Manager->dispatch('clean_cache_by_...', Array)
#18 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/Model/AbstractModel.php(801):
Magento\Framework\Event\Manager\Proxy->dispatch('clean_cache_by_...', Array)
#19 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/Model/ResourceModel/Db/AbstractDb.php(818):
Magento\Framework\Model\AbstractModel->afterSave()
#20 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/Model/ResourceModel/Db/AbstractDb.php(408):
Magento\Framework\Model\ResourceModel\Db\AbstractDb->processAfterSaves(Object(Magestore\Webpos\Model\Staff\WebPosSession))
#21 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/Model/AbstractModel.php(631):
Magento\Framework\Model\ResourceModel\Db\AbstractDb->save(Object(Magestore\Webpos\Model\Staff\WebPosSession))
#22 /var/www/vhosts/
ooples.com/httpdocs/app/code/Magestore/Webpos/Helper/Permission.php(155):
Magento\Framework\Model\AbstractModel->save()
#23 /var/www/vhosts/
ooples.com/httpdocs/app/code/Magestore/Webpos/Controller/Rest.php(42):
Magestore\Webpos\Helper\Permission->authorizeSession('tvqgrm09qkb5gp5...')
#24 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/module-webapi/Controller/Rest.php(334):
Magestore\Webpos\Controller\Rest->checkPermissions()
#25 /var/www/vhosts/
ooples.com/httpdocs/app/code/Magestore/Webpos/Controller/Rest.php(94):
Magento\Webapi\Controller\Rest->validateRequest()
#26 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/module-webapi/Controller/Rest.php(216):
Magestore\Webpos\Controller\Rest->processApiRequest()
#27 /var/www/vhosts/
ooples.com/httpdocs/var/generation/Magestore/Webpos/Controller/Rest/Interceptor.php(63):
Magento\Webapi\Controller\Rest->dispatch(Object(Magento\Framework\App\Request\Http))
#28 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/App/Http.php(135):
Magestore\Webpos\Controller\Rest\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#29 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/App/Bootstrap.php(258):
Magento\Framework\App\Http->launch()
#30 /var/www/vhosts/ooples.com/httpdocs/index.php(39):
Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#31 {main}

Next Exception: Report ID: webapi-58e55d19cb294; Message: Warning:
json_decode() expects parameter 1 to be string, object given in
/var/www/vhosts/
ooples.com/httpdocs/vendor/cloudflare/cloudflare-plugin-backend/src/Integration/DefaultConfig.php
on line 14 in /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/Webapi/ErrorProcessor.php:195
Stack trace:
#0 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/Webapi/ErrorProcessor.php(139):
Magento\Framework\Webapi\ErrorProcessor->_critical(Object(Exception))
#1 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/module-webapi/Controller/Rest.php(219):
Magento\Framework\Webapi\ErrorProcessor->maskException(Object(Exception))
#2 /var/www/vhosts/
ooples.com/httpdocs/var/generation/Magestore/Webpos/Controller/Rest/Interceptor.php(63):
Magento\Webapi\Controller\Rest->dispatch(Object(Magento\Framework\App\Request\Http))
#3 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/App/Http.php(135):
Magestore\Webpos\Controller\Rest\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#4 /var/www/vhosts/
ooples.com/httpdocs/vendor/magento/framework/App/Bootstrap.php(258):
Magento\Framework\App\Http->launch()
#5 /var/www/vhosts/ooples.com/httpdocs/index.php(39):
Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#6 {main} [] []

**Comment by jwineman** [comment]
Thats the same error as before.  Lets make sure the plugin is actually up to date:


1. Delete that line (`"cloudflare/cloudflare-magento": "^1.0"`) in your `composer.json`
2. Run `composer update`
3. Run `composer require cloudflare/cloudflare-magento`

**Comment by ooples** [comment]
I believe the update worked properly the first time because I was seeing
"cloudflare/cloudflare-magento": "^1.1", now in the composer.json file and
when I ran composer update from earlier, it said it installed cloudflare
1.1.3

**Comment by jwineman** [comment]
I know, I want to be sure though.

**Comment by ooples** [comment]
Ok I redid all steps and cleaned my cache several times and it is giving me
that same error message which I have attached. This is my second time
confirming the steps

**Comment by ooples** [comment]
Do you have any update on this issue?

**Comment by jwineman** [comment]
Yup - finalizing a fix as we speak.

**Comment by ooples** [comment]
Any update?

**Comment by jwineman** [comment]
Hey - sorry.  I released 1.1.4 which fixes the JSON issue.  I'm currently talking to the Magento team debugging an issue where `setup:di:compile` complains that `RequestRouter` must implement `RouterInterface` even though it does.  If I remove the `var/` folder after running `setup:di:compile` and load the page it generates correctly.  

I would actually be curious if you're getting this as well - does 1.1.4 work for you?

**Comment by ooples** [comment]
Here are the steps I followed to get 1.1.4 installed without any errors:

composer update
composer require cloudflare/cloudflare-magento
php -f bin/magento module:enable CloudFlare_Plugin
php -f bin/magento setup:upgrade
rm -rf var/generation/*
rm -rf var/cache/*
php -dmemory_limit=15G -f bin/magento setup:di:compile
rm -rf pub/static/*
rm -rf var/view_preprocessed/*
php -dmemory_limit=15G -f bin/magento setup:static-content:deploy

However after installation, I entered in my email and api key and hit save
api credentials and nothing seems to happen. I cleared cache and cookies
and tried again and still nothing :(

**Comment by jwineman** [comment]
Yeah that is the same error I'm getting.  The AJAX request is 500ing.  I bet if you remove the `var/` folder it will fix itself.

**Comment by ooples** [comment]
Ok I can confirm that if you remove the var folder then it starts working

**Comment by jwineman** [comment]
Good!  Hopefully this is a good work around for now.  I'll update this thread again when I hear back from the Magento QA team.

**Comment by ooples** [comment]
I was under the impression that this extension provided a fix for magento
not being able to see the true ip of visitors because of cloudflare. Is
that not correct or a possibility for this extension?

**Comment by jwineman** [comment]
Correct - if you have the extension installed [this code](https://github.com/cloudflare/Cloudflare-Magento/blob/master/etc/di.xml#L48-L55) will provide the real IP address of the visitor to Magento.

**Comment by jwineman** [comment]
Hi @ooples,

This is completely fixed in [1.1.5](https://github.com/cloudflare/Cloudflare-Magento/releases/tag/v1.1.5)!

Thanks,
John

**Comment by ooples** [comment]
Good to hear
